compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/SwitchGenerator.kt

```diff
@@ -43,36 +43,72 @@ class SwitchGenerator(private val expression: IrWhen, private val data: BlockInf
             }
         }
 
-        // IF is more compact when there are only 1 or fewer branches, in addition to else.
-        if (allConditions.size <= 1)
+        // switch isn't applicable if there's no case at all, e.g., when() { else -> ... }
+        if (allConditions.size == 0)
             return null
 
-        if (areConstIntComparisons(allConditions.map { it.first })) {
+        // Remove labels and "then expressions" that are not reachable.
+        fun removeUnreachableExpressions(reachables: HashSet<Label>) = thenExpressions.removeIf { it.second !in reachables }
+
+        val calls = allConditions.map { it.first }
+        if (areConstIntComparisons(calls)) {
```

Comment:
Branches seem identical except return value: maybe merge then with explicit cast for cases in each constructor

Reply:
Done. `areConstComparisons` are hoisted out of `areConst[type]Comparisons` to guard the shared operations, which are moved before `areConst[type]Comparisons`.